cmd/gcp-controller-manager/node_csr_approver_test.go

```diff
@@ -1640,3 +1644,126 @@ func TestParseInstanceGroupURL(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteAllPodsBoundToNode(t *testing.T) {
+	testNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{Name: "testNode"},
+		Status: v1.NodeStatus{
+			Capacity: v1.ResourceList{
+				v1.ResourceName(v1.ResourceCPU):    resource.MustParse("10"),
+				v1.ResourceName(v1.ResourceMemory): resource.MustParse("10G"),
+			},
+		},
+	}
+
+	testPod := &v1.Pod{
```

Comment:
The test does not cover the execution path when the pod is already in a terminal phase, or when PodDisruptionConditions is enabled. I'm wondering about having a test case table to check these scenarios.

Reply:
I added additional test cases, ptal.